### PR TITLE
313024-311852-311851-311850 - Date of testing logics with it's except…

### PR DESCRIPTION
**Pull request by Afshin-Karamifar-Capgemini**
…ion screens

## Checklist

- [x] Content has no typos
- [x] Correct inline validation errors
- [x] Back link and next page navigation
- [x] Removed all unnecessary console logs
- [x] Removed all commented out code
- [ ] Updated README / confluence

### CODE REVIEW

File: app/routes/endemics/date-of-testing.js [hunk 1]
```diff
@@ -189,15 +190,29 @@ module.exports = [
...
           )
+
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.review) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionOne' }).code(400).takeover()
+        }
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.endemics) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionTwo' }).code(400).takeover()
+        }
+
+        const previousReviewClaim = getReviewWithinLast10Months(dateOfVisit, previousClaims)
```
Comment (alex742) [inline app/routes/endemics/date-of-testing.js:209]: Need to also pass in the old world claim (which might be undefined)

File: app/routes/endemics/date-of-testing.js [hunk 2]
```diff
@@ -189,15 +190,29 @@ module.exports = [
...
             request.payload['on-another-date-year'],
             request.payload['on-another-date-month'] - 1,
             request.payload['on-another-date-day']
           )
+
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.review) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionOne' }).code(400).takeover()
+        }
+        if (!isWithIn4MonthsBeforeOrAfterDateOfVisit(dateOfVisit, dateOfTesting) && typeOfReview === claimType.endemics) {
+          return h.view(endemicsDateOfTestingException, { backLink: pageUrl, ruralPaymentsAgency, exception: 'exceptionTwo' }).code(400).takeover()
```
Comment (alex742) [inline app/routes/endemics/date-of-testing.js:206]: Can you rename 'exceptionOne', 'exceptionTwo', and 'exceptionThree' to something more meaningful?

File: test/unit/app/api-requests/claim-service-api.test.js [hunk 3]
```diff
@@ -76,4 +76,28 @@ describe('Claim Service API', () => {
...
+    const { isWithIn4MonthsAfterDateOfVisit } = require('../../../../app/api-requests/claim-service-api')
+
+    expect(isWithIn4MonthsAfterDateOfVisit(new Date('2024-04-23'), new Date('2024-06-23'))).toBe(true)
+    expect(isWithIn4MonthsAfterDateOfVisit(new Date('2024-04-23'), new Date('2024-10-23'))).toBe(false)
+  })
+  test('Check if date of visit is valid for when type of review is not review or endemics', async () => {
+    const { isValidDateOfVisit } = require('../../../../app/api-requests/claim-service-api')
+
+    expect(isValidDateOfVisit('T')).toMatchObject({ isValid: false })
+  })
```
Comment (alex742) [inline test/unit/app/api-requests/claim-service-api.test.js:102]: Can you also test for edge cases?
Exactly 4 months, 4 months and 1 day, 1 day less than 4 months

**Review by alex742 — APPROVED** [review]
